### PR TITLE
Docs: Mock Phone Number

### PR DESCRIPTION
**Pull request by choir27**
## What does this PR do?

Add section to document mocking phone numbers, what it does for the user, and how to mock the number.
Adds notes and links to the mock phone number docs sections in relevant docs

## Test Plan

Manually tested mocking a phone number in the preview instance Cloud

## Related PRs and Issues

X

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes

### CODE REVIEW

File: src/routes/docs/advanced/self-hosting/sms/+page.markdoc [hunk 1]
```diff
@@ -4,7 +4,7 @@ title: SMS delivery
 description: Set up SMS services for your self-hosted Appwrite instance. Discover how to configure SMS notifications, verification, and messaging for your applications.
 ---
 
-Appwrite supports phone authentication, which allows users to create accounts and log in using SMS messages. Appwrite requires an SMS provider to be set up before using Phone authentication.
+Appwrite supports phone authentication, which allows users to create accounts and log in using SMS messages. Appwrite requires an SMS provider to be set up before using Phone authentication; however, you can also mock the SMS provider by mocking the [phone number](/docs/products/auth/security#mock-phone-number).
```
Comment (stnguyen90) [inline src/routes/docs/advanced/self-hosting/sms/+page.markdoc:7]: Let's remove this for now.

File: src/routes/docs/products/auth/phone-sms/+page.markdoc [hunk 2]
```diff
@@ -98,7 +98,7 @@ mutation {
 
 # Login {% #login %}
 
-After initiating the phone authentication process, the returned user ID and secret are used to confirm the user. The secret will usually be a 6-digit number in the SMS message sent to the user.
+After initiating the phone authentication process, the returned user ID and secret are used to confirm the user. The secret will usually be a 6-digit number in the SMS message sent to the user. You can also [mock the phone number](/docs/products/auth/security#mock-phone-number) to initiate a phone authentication process.
```
Comment (stnguyen90) [inline src/routes/docs/products/auth/phone-sms/+page.markdoc:101]: This is a little hidden. Let's put a paragraph before the "Send SMS message' heading.

Also, we're not mocking phone numbers. "Mock" is an adjective.

File: src/routes/docs/products/auth/security/+page.markdoc [hunk 3]
```diff
@@ -21,6 +21,17 @@ Only keep user sessions active as long as needed and maintain exactly **one** in
 | {% only_dark %}{% icon_image src="/images/platforms/dark/apple.svg" alt="Javascript logo" size="m" /%}{% /only_dark %}{% only_light %}{% icon_image src="/images/platforms/apple.svg" alt="Javascript logo" size="m" /%}{% /only_light %}  | Apple | Uses a session cookie stored in **UserDefaults**. |
 | {% only_dark %}{% icon_image src="/images/platforms/dark/android.svg" alt="Javascript logo" size="m" /%}{% /only_dark %}{% only_light %}{% icon_image src="/images/platforms/android.svg" alt="Javascript logo" size="m" /%}{% /only_light %}  | Android | Uses a session cookie stored in **SharedPreferences**. |
 
+# Mock phone number {% #mock-phone-number %}
+
+Mocking phone numbers allows users to test SMS authentication without needing an actual phone number. This can be useful for testing edge cases where a user doesn't have a physical SIM card but needs to sign in to your application using SMS. To mock a phone number, navigate to **Auth** > **Security** > Mock Phone Numbers.
```
Comment (stnguyen90) [inline src/routes/docs/products/auth/security/+page.markdoc:26]: "Mocking phone numbers" doesn't quite sound right since "Mock" is an adjective.

> physical SIM card

Let's not say this because you can still receive SMS without a physical SIM card.

This background/context/detail is good. 

Please clearly explain what happens when using a mock number.

File: static/images/docs/auth/phone/dark/mock-phone.png
Comment (stnguyen90) [inline static/images/docs/auth/phone/dark/mock-phone.png:1]: Please make sure to follow [style guidelines](https://github.com/appwrite/website/blob/main/STYLE.md#screenshots) for screenshots:

- Screenshots must be 16:9
- Screenshots should be taken in a 1400 x 900 view port on 3x DPR in browser developer tools.
- Use generic and sensible organization, project, and resource names. Avoid names like `test`, `demo`, or `sdlkfj`.
- All screenshot should be take from a user named Walter O'Brien. You can change the name of your current user by going to your Appwrite Console and clicking the **top right profile icon** > **Your Account** > **Name**.

**Review by stnguyen90 — CHANGES_REQUESTED** [review]

File: src/routes/docs/products/auth/phone-sms/+page.markdoc (hunk 2 above)
Comment (choir27) [reply src/routes/docs/products/auth/phone-sms/+page.markdoc:101]: But the Console in the preview instance currently words it as Mock Phone Numbers, using "Mock" as a verb. Shouldn't we stay consistent with the Console? Or is this a mistake in the Console?

Comment (stnguyen90) [reply src/routes/docs/products/auth/phone-sms/+page.markdoc]: > Console in the preview instance

This?

image

Mock can either be a verb or adjective. In this case, it's an adjective.

File: src/routes/docs/products/auth/security/+page.markdoc [hunk 4]
```diff
@@ -21,6 +21,19 @@ Only keep user sessions active as long as needed and maintain exactly **one** in
...
+Using mock phone numbers allows users to test SMS authentication without needing an actual phone number. This can be useful for testing edge cases where a user doesn't have a phone number but needs to sign in to your application using SMS. 
+
+After defining a mock phone number, you need to define a specific OTP code to use for SMS sign-in instead of the SMS secret code sent to a real phone number. To use a mock a phone number, navigate to **Auth** > **Security** > Mock Phone Numbers.
+
+{% only_dark %}
+![Console UI to mock phone number](/images/docs/auth/phone/dark/mock-phone.png)
+{% /only_dark %}
+{% only_light %}
+![Console UI to mock phone number](/images/docs/auth/phone/mock-phone.png)
+{% /only_light %}
```
Comment (stnguyen90) [inline src/routes/docs/products/auth/security/+page.markdoc:35]: Actually, sorry. There's no screenshot for the others so let's leave it out for now. It's weird to have a screenshot just for this one.

File: src/routes/docs/products/auth/security/+page.markdoc [hunk 5]
```diff
@@ -21,6 +21,19 @@ Only keep user sessions active as long as needed and maintain exactly **one** in
 | {% only_dark %}{% icon_image src="/images/platforms/dark/apple.svg" alt="Javascript logo" size="m" /%}{% /only_dark %}{% only_light %}{% icon_image src="/images/platforms/apple.svg" alt="Javascript logo" size="m" /%}{% /only_light %}  | Apple | Uses a session cookie stored in **UserDefaults**. |
 | {% only_dark %}{% icon_image src="/images/platforms/dark/android.svg" alt="Javascript logo" size="m" /%}{% /only_dark %}{% only_light %}{% icon_image src="/images/platforms/android.svg" alt="Javascript logo" size="m" /%}{% /only_light %}  | Android | Uses a session cookie stored in **SharedPreferences**. |
 
+# Using mock phone number {% #using-mock-phone-number %}
+
+Using mock phone numbers allows users to test SMS authentication without needing an actual phone number. This can be useful for testing edge cases where a user doesn't have a phone number but needs to sign in to your application using SMS. 
+
+After defining a mock phone number, you need to define a specific OTP code to use for SMS sign-in instead of the SMS secret code sent to a real phone number. To use a mock a phone number, navigate to **Auth** > **Security** > Mock Phone Numbers.
```
Comment (stnguyen90) [inline src/routes/docs/products/auth/security/+page.markdoc:28]: > OTP code to use for SMS sign-in instead of the SMS secret code sent to a real phone number.

This was the important part that needed to be conveyed. However, let's swap the order of the sentences so that the directions are to go to the Security tab first, then define the mock number and OTP that will let you sign in instead of using a real phone number.

You use a mock number in the app. You don't use a mock number in the Console. Instead, you *create* a mock number.

**Review by stnguyen90 — CHANGES_REQUESTED** [review]

File: src/routes/docs/products/auth/security/+page.markdoc [hunk 6]
```diff
@@ -21,6 +21,19 @@ Only keep user sessions active as long as needed and maintain exactly **one** in
 | {% only_dark %}{% icon_image src="/images/platforms/dark/apple.svg" alt="Javascript logo" size="m" /%}{% /only_dark %}{% only_light %}{% icon_image src="/images/platforms/apple.svg" alt="Javascript logo" size="m" /%}{% /only_light %}  | Apple | Uses a session cookie stored in **UserDefaults**. |
 | {% only_dark %}{% icon_image src="/images/platforms/dark/android.svg" alt="Javascript logo" size="m" /%}{% /only_dark %}{% only_light %}{% icon_image src="/images/platforms/android.svg" alt="Javascript logo" size="m" /%}{% /only_light %}  | Android | Uses a session cookie stored in **SharedPreferences**. |
 
+# Using mock phone number {% #using-mock-phone-number %}
```
Comment (stnguyen90) [inline src/routes/docs/products/auth/security/+page.markdoc:24]: Header should just be "Mock Phone Numbers"

File: src/routes/docs/products/auth/phone-sms/+page.markdoc [hunk 7]
```diff
@@ -6,6 +6,8 @@ description: Enhance security with phone and SMS authentication in Appwrite. Dis
 
 Phone authentication lets users create accounts using their phone numbers and log in through SMS messages.
 
+Use [mock phone numbers](/docs/products/auth/security#mock-phone-number) to initiate a phone authentication process without an actual phone number.
```
Comment (stnguyen90) [inline src/routes/docs/products/auth/phone-sms/+page.markdoc:9]: Make sure the fragment matches the ID on the page.

**Review by stnguyen90 — APPROVED** [review]
